pkg/config/model.go

```diff
@@ -128,7 +129,7 @@ func ToProto(s Spec) (*anypb.Any, error) {
 	// golang/protobuf 1.4+ will have this interface. Older golang/protobuf are gogo compatible
 	// but also not used by Istio at all.
 	if pb, ok := s.(protoreflect.ProtoMessage); ok {
-		return anypb.New(pb)
+		return protoconv.MessageToAny(pb), nil
```

Comment:
why not `MessageToAnyWithError`, so we would not miss the err value

Reply:
`MessageToAny` will call `MessageToAnyWithError`, and it will return error if failed. But what you said makes sense, here we can directly call the `MessageToAnyWithError `function, thx.

Reply:
Fixed.